x/interchaintxs/keeper/msg_server.go

```diff
@@ -47,8 +47,10 @@ func (k Keeper) RegisterInterchainAccount(goCtx context.Context, msg *ictxtypes.
 		return nil, errors.Wrapf(ictxtypes.ErrNotContract, "%s is not a contract address", msg.FromAddress)
 	}
 
-	if err := k.ChargeFee(ctx, senderAddr, msg.RegisterFee); err != nil {
-		return nil, errors.Wrapf(err, "failed to charge fees to pay for RegisterInterchainAccount msg: %s", msg)
+	if k.sudoKeeper.GetContractInfo(ctx, senderAddr).CodeID >= k.GetLastCodeIDBeforeUpgrade(ctx) {
```

Comment:
why `>=`? we will charge the contracts with src code uploaded before upgrade this way, won't we? seems like here a strict `>` is required